### ISSUE TITLE
Wrong state of the plug after a short power outage

### ISSUE THREAD

**Issue by YuriiMaiboroda** [make-all/tuya-local#377]
Hello,

I have a wifi plug (sold as "Tuya Smart WiFi Elivco 20A", "ELIVCO LSPA9" is written on the plug, in the information about the device from api.tuya it is written that the model is "CBS2S-16DL").

When connected to Tuya Local, it was defined as `smartplugv2`, but in this case, not all controls are shown, so I slightly modified `smartplugv2_energyv3` to fit my plug.
<details>
  <summary>smart_plugv2_energyv4.yaml</summary>

  ```yaml
name: Energy monitoring smartplug
primary_entity:
  entity: switch
  class: outlet
  dps:
    - id: 1
      type: boolean
      name: switch
    - id: 21
      type: integer
      name: test_bit
    - id: 22
      type: integer
      name: voltage_calibration
    - id: 23
      type: integer
      name: current_calibration
    - id: 24
      type: integer
      name: power_calibration
    - id: 25
      type: integer
      name: energy_calibration
    - id: 26
      type: bitfield
      name: fault_code
    - id: 42
      type: string
      name: cycle_time
    - id: 43
      type: string
      name: random_time
    - id: 44
      type: string
      name: switch_inching
secondary_entities:
  - entity: number
    category: config
    name: Timer
    icon: "mdi:timer"
    dps:
      - id: 9
        type: integer
        name: value
        unit: sec
        range:
          min: 0
          max: 86400
        mapping:
          - scale: 1
            step: 1
  - entity: sensor
    category: diagnostic
    class: current
    name: Current
    dps:
      - id: 18
        name: sensor
        type: integer
        class: measurement
        unit: mA
        mapping:
          - scale: 1
            step: 1
  - entity: sensor
    category: diagnostic
    class: power
    name: Power
    dps:
      - id: 19
        name: sensor
        type: integer
        class: measurement
        unit: W
        mapping:
          - scale: 10
            step: 1
  - entity: sensor
    category: diagnostic
    class: voltage
    name: Voltage
    dps:
      - id: 20
        name: sensor
        type: integer
        class: measurement
        unit: V
        mapping:
          - scale: 10
            step: 1
  - entity: binary_sensor
    class: problem
    category: diagnostic
    name: Error
    dps:
      - id: 26
        type: bitfield
        name: sensor
        mapping:
          - dps_val: 0
            value: false
          - value: true
  - entity: select
    category: config
    name: Initial state
    icon: "mdi:toggle-switch"
    dps:
      - id: 38
        type: string
        name: option
        mapping:
          - dps_val: "on"
            value: "On"
          - dps_val: "off"
            value: "Off"
          - dps_val: "last"
            value: "Last State"
  - entity: switch
    name: Overcharge switch
    category: config
    icon: "mdi:toggle-switch"
    dps:
      - id: 39
        type: boolean
        name: switch
  - entity: select
    name: Light mode
    category: config
    icon: "mdi:lightbulb"
    dps:
      - id: 40
        type: string
        name: option
        mapping:
          - dps_val: "relay"
            value: "State"
          - dps_val: "pos"
            value: "Position"
          - dps_val: "none"
            value: "Off"
          - dps_val: "on"
            value: "On"
  - entity: lock
    name: Child Lock
    category: config
    icon: "mdi:hand-back-right-off"
    dps:
      - id: 41
        type: boolean
        name: lock
  ```
 [smart_plugv2_energyv4.txt](https://github.com/make-all/tuya-local/files/10697748/smart_plugv2_energyv4.txt)
</details>

Everything works, it seems, well, except for restoring the state after short power failure in the electrical network.

I have it set up so that the plug remains OFF after power in electrical network is restored, even if it was turned ON before the power went out.
If the power is lost for 5 minutes, the device becomes unavailable and after the power is restored it is restored as expected - the plug is recognized as turned off.
But if the power goes out for  les then 5 minutes (e.g. for 10 seconds), then the plug will turn OFF, but in Tuya Local it continues to be considered ON (as it was before the loss of power).

Is there anything else I can set up to get the current state of the plug, and not the one that was before the short power outage?

Ideally, if there is a power outage, the plug goes into the "unavailable" state immediately, and not after 5 minutes.

**Comment by YuriiMaiboroda** [comment]
I was wrong. It turns out Tuya Local still detects that the plug is disabled, but after a while. I noticed a change in state after a time from a few seconds to 3 minutes.
Is it possible to somehow shorten this time so that it is always a few seconds?

**Comment by make-all** [comment]
You can likely shorten the detection time by explicitly setting the protocol version and IP address.  If you have used "Auto" for either of those the integration may be spending a fair amount of time trying to find which protocol version works.

**Comment by YuriiMaiboroda** [comment]
I already set a specific IP address and protocol 3.3 during the initial setup.

**Comment by YuriiMaiboroda** [comment]
Is it possible to configure it so that if, when polling devices, it does not respond, immediately consider it unavailable? Instead of doing 5 retries for each request for 5 minutes.
Or will I need to edit the tinytuya source code for this?

**Comment by YuriiMaiboroda** [comment]
I also noticed that the data from the power, voltage and current sensors (18, 19, 20) are rarely updated unless the Tuya Smart application is open.
If the application is open - the data is updated approximately every 2 seconds. If closed, it may not be updated for 20-30 minutes, and sometimes more.
In the log from tinytuya.core, approximately every 2 minutes, in addition to the heartbeat, an output appears with a complete list of DPS, but each time there are the same values. This behavior seemed to me similar to the behavior of a device without power - the data is also not updated in any way, despite the fact that requests are coming.

In the tinytuya API, I found the `updatedps` function, which was recommended to be used [here](https://github.com/tuya/tuya-home-assistant/issues/211).
Perhaps if you call it in addition to getting the status, then the data will come more stably, and maybe it will be possible to quickly determine that the device is unavailable?
